src/main/java/org/programmers/springboot/basic/domain/voucher/service/VoucherService.java

```diff
@@ -0,0 +1,125 @@
+package org.programmers.springboot.basic.domain.voucher.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.programmers.springboot.basic.domain.voucher.dto.VoucherRequestDto;
+import org.programmers.springboot.basic.domain.voucher.dto.VoucherResponseDto;
+import org.programmers.springboot.basic.domain.voucher.entity.Voucher;
+import org.programmers.springboot.basic.domain.voucher.entity.VoucherType;
+import org.programmers.springboot.basic.domain.voucher.exception.DuplicateVoucherException;
+import org.programmers.springboot.basic.domain.voucher.exception.VoucherNotFoundException;
+import org.programmers.springboot.basic.domain.voucher.mapper.VoucherMapper;
+import org.programmers.springboot.basic.util.generator.DateTimeGenerator;
+import org.programmers.springboot.basic.util.generator.UUIDGenerator;
+import org.programmers.springboot.basic.domain.voucher.repository.VoucherRepository;
+import org.programmers.springboot.basic.domain.voucher.service.validate.VoucherValidationStrategy;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+@Profile("default")
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class VoucherService {
+
+    private final VoucherRepository voucherRepository;
+    private final UUIDGenerator uuidGenerator;
+    private final DateTimeGenerator dateTimeGenerator;
+    private final Map<VoucherType, VoucherValidationStrategy> validationStrategyMap;
+
+    @Transactional
+    public void create(VoucherRequestDto voucherRequestDto) {
+
+        Long discount = voucherRequestDto.getDiscount();
+        VoucherType voucherType = voucherRequestDto.getVoucherType();
+
+        Voucher voucher = VoucherMapper.INSTANCE.mapToEntityWithGenerator(voucherRequestDto,
+                                                                        uuidGenerator,
+                                                                        dateTimeGenerator,
+                                                                        validationStrategyMap);
+        voucher.validate();
```

Comment:
```suggestion
        VoucherValidationStrategy validationStrategy = validationStrategyMap.get(voucherRequestDto.getVoucherType());
        if (validationStrategy == null) {
            throw new IllegalStateException("바우처 타입 전략 찾을 수 없는 예외: " + voucherRequestDto.getVoucherType());
        }

        Voucher voucher = VoucherMapper.INSTANCE.mapToEntityWithGenerator(
                voucherRequestDto,
                uuidGenerator,
                dateTimeGenerator,
                validationStrategyMap
        );
        voucher.setValidationStrategy(validationStrategy);
        voucher.validate();
```

작성하신 코드에서 어떤게 필요한거였을까요?